artifact/v3/header/type_info.cpp

```diff
@@ -64,18 +64,24 @@ ExpectedTypeInfo Parse(io::Reader &reader) {
 
 	log::Trace("type-info: Parsing the payload type");
 
-	auto expected_payload = type_info_json.Get("type").and_then(json::ToString);
+	auto expected_payload = type_info_json.Get("type");
 	if (!expected_payload) {
+		return expected::unexpected(parser_error::MakeError(
+			parser_error::Code::ParseError,
+			"Failed to get the type-info payload type JSON: " + expected_payload.error().message));
+	}
+	auto payload_type = expected_payload.value();
+	if (payload_type.IsNull()) {
+		type_info.type = "null";
```

Comment:
This overlaps with the potential real name `null`, no? We should probably use an extra enum or something to track this.

Reply:
Discussed f2f, resolving independently